auth_matrix/admin_views.py

```diff
@@ -1,15 +1,37 @@
+from typing import Any
+
 from django import views
-from django.contrib.admin.views.decorators import staff_member_required
 from django.contrib.auth import get_user_model
+from django.contrib.auth.mixins import UserPassesTestMixin
 from django.contrib.auth.models import Group, Permission
-from django.utils.decorators import method_decorator
+from django.http import HttpRequest
+from django.http.response import HttpResponse
 
 User = get_user_model()
 
-@method_decorator(staff_member_required, name="dispatch")
-class AuthorizationMatrixView(views.generic.base.TemplateView):
 
-    template_name = "admin/authorization/authorization_matrix.html"
+class UserIsStaffMixin(UserPassesTestMixin):
+    def test_func(self):
+        return self.request.user.is_staff
+
+
+def get_auth_matrix_permission_label():
+    opts = User._meta
+    return f"{opts.app_label}.view_{opts.model_name}"
+
+
+class CanViewAuthorizationMatrixMixin(UserIsStaffMixin):
+    def dispatch(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
+        if not request.user.has_perm(get_auth_matrix_permission_label()):
+            return self.handle_no_permission()
+        return super().dispatch(request, *args, **kwargs)
+
+
+class AuthorizationMatrixView(
+    CanViewAuthorizationMatrixMixin,
+    views.generic.base.TemplateView,
+):
+    template_name = "auth_matrix/admin/authorization_matrix.html"
 
     def get(self, request):
         groups = Group.objects.prefetch_related("permissions").order_by("name")
```

Comment:
I think this has the same issues as the export w/r to the permission names not being unique and filtering on name especially may show an auth matrix that is wrong.

It's a good exercise for you to write a single `Group` or `Permission` query that already includes the necessary information via `prefetch` so that you can avoid these problems :)

Reply:
Here we are ordering groups by name so that should not be an issue since they are unique.
But I understand the concern for permissions. I have submitted the changes.

Reply:
right I had some PR review blindness here, it's already using the prefetch so you can disregard my remark. It's hard to review code that's building on top of unreviewed code.